StripeCore/StripeCore/Source/Analytics/AnalyticLoggableError.swift

```diff
@@ -65,4 +68,69 @@ where Self: RawRepresentable, Self.RawValue == String {
         }
     }
 
+    /// Serialize an Error for logging to q.stripe.com and the `sdk.analytics_events` table
+    ///
+    /// It sends the following fields:
+    /// - error_type: For Stripe API errors, the error’s [type](https://docs.stripe.com/api/errors#errors-type) e.g. “invalid_request_error”.
+    ///           For Swift errors, the fully qualified type name e.g. “StripePaymentSheet.LinkURLGeneratorError”.
+    ///           For NSErrors, the error domain e.g. “NSURLErrorDomain”.
+    /// - error_code: For Stripe API errors, the error's code e.g. "invalid_number".
+    ///            For NSErrors, the error code e.g. “-1009”.
+    ///            For Swift errors, the enum case name as a string for Swift errors e.g. “noPublishableKey”.
+    public func serializeForV1Analytics() -> [String: Any] {
+        let errorType = Self.extractErrorType(from: self)
+        let errorCode = Self.extractErrorCode(from: self)
+
+        return [
+            "error_type": errorType,
+            "error_code": errorCode,
+        ]
+    }
+
+    /// Extracts a value suitable for the `"error_type"` analytic parameter
+    /// - For Stripe API errors, the error’s [type](https://docs.stripe.com/api/errors#errors-type) e.g. “invalid_request_error”.
+    /// - For Swift errors, the fully qualified type name e.g. “StripePaymentSheet.LinkURLGeneratorError”.
+    /// - For NSErrors, the error domain e.g. “NSURLErrorDomain”.
+    static func extractErrorType(from error: Error) -> String {
+        if type(of: error) is NSError.Type {
+            let error = error as NSError
```

Comment:
Does this if statement catch anything that `if let error = error as? NSError` wouldn't?

Reply:
Oh this code's comments need to be improved, thanks for bringing it up.

Casting an Error to NSError always succeeds - Swift automatically bridges them. This line checks that the error is an actual instance of NSError.